src/utils/yupvalidators.js

```diff
@@ -0,0 +1,34 @@
+import * as yup from 'yup';
+
+const min = 8;
+const max = 25;
+
+export const signUpSchema = yup.object().shape({
+  confirmPassword: yup
+    .string()
+    .required('this field is required')
+    .min(min, 'It must have 8 characters or more')
+    .max(max, 'It must have 25 characters or less')
+    .matches(
+      /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$/,
+      'Password must have contain at least one mayus letter and a number'
+    )
+    .when('password', (password, schema) => {
+      schema.test('confirm', 'Password should match!', value => value === password);
+    }),
+  email: yup
+    .string()
+    .email('email not valid')
+    .required('this field is required'),
+  firstName: yup.string().required('this field is required'),
+  lastName: yup.string().required('this field is required'),
+  password: yup
+    .string()
+    .required('this field is required')
+    .min(min, 'It must have 8 characters or more')
+    .max(max, 'It must have 25 characters or less')
+    .matches(
+      /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$/,
+      'Password must have contain at least one mayus letter and a number'
+    )
+});
```

Comment:
this is so overkill and could be done without yup using just the register and the field level validation that comes with react-hook-form.

yup works at form level validation, but its better to use field level validation.

For example for the password:
```
ref={register({
    required: i18next.t('FormValidations:required'),
    minLength: {
      value: 6,
      message: i18next.t('FormValidations:minLength', {
        min: 6
      })
    }
  })}
```
for the email i recommend you this regex
```
export const emailRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```

and to use it is like

```
ref={register({
    required: i18next.t('FormValidations:required'),
    pattern: {
      value: emailRegex,
      message: i18next.t('FormValidations:email')
    }
  })
}
```

and in the form. you pass. the error object to the component with the object `errors` with the key of the field (i.e: `errors['email']`) exactly as you were doing.

Also is a good practice to keep a `fields.js` file to store all the field names. Example:

```
export const AUTH_FIELDS = {
email: 'email',
password: 'password',
confirmPassword: 'confirmPassword',
firstName: 'firstName',
lastName: 'lastName'
}
```

Reply:
Okay, if you're used to using it like that, I'll change it.

But with yup I think that the code is much cleaner and adds other features that can be used at the time of validation.